src/App.tsx

```diff
@@ -1,61 +1,140 @@
+import React, { FC, useState } from 'react';
+import { User, Todo } from './react-app-env';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { getNewId } from './helpers/helpers';
+
+const getUserById = (userId: number): User | null => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [
+    titleErrorMessage,
+    setTitleErrorMessage,
+  ] = useState('');
+  const [
+    userErrorMessage,
+    setUserErrorMessage,
+  ] = useState('');
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title || !userId) {
+      setTitleErrorMessage(!title ? 'Please enter a title' : '');
+      setUserErrorMessage(!userId ? 'Please choose a user' : '');
+
+      return;
+    }
+
+    setTodos(prev => {
+      const newTodo = {
+        id: getNewId(prev),
+        title,
+        completed: false,
+        user: getUserById(userId),
+      };
+
+      return [...prev, newTodo];
+    });
+
+    reset();
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleErrorMessage('');
+    setTitle(event.target.value);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserErrorMessage('');
+    setUserId(+event.target.value);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">
+            {'Title: '}
+          </label>
+
+          <input
+            id="titleInput"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={(event) => handleInputChange(event)}
```

Comment:
If your method get event as argument or don't get any argument, you don't need to create new arrow function. Just pass your method.